src/main/webapp/js/angularApp/controllers.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+/* global kanban, _, $, angular */
+
+kanban.controller('kanbanController', ['Issues', '$rootScope', '$scope', '$timeout', function(Issues, $rootScope, $scope, $timeout) {
+
+  //default request - sending a click to the TL Staff button
+  $timeout(function() {
+    angular.element('#tlstaffbutton').triggerHandler('click');
+  }, 0);
+  // what collection is being shown is stored in the root scope
+  // not used now but might come in handy later
+  //var issuesUrl = '/jirarequest?' + $rootScope.sourceCollection;
+  
+  var wipUrl='/jirarequest?wip';
+  
+  // use a factory as a promise to get WIP data
+  Issues.getWip(wipUrl).then(function(result) {
+    if (result.data.errors) {
+      //source error: do something
+    } else {
+      if (result.errors) {
+        //angular error: do something else
+      } else {
+        $scope.wip = result.data;
+      }
+    }
+  });
+
+  // click handler for subsequent requests via buttons
+  $scope.switchIssueCollection = function(collectionParam) {
+    
+    // start the spinner
+    $scope.loading = true;
+    
+    var issuesUrl = '/jirarequest?' + collectionParam;
+    $rootScope.sourceCollection = collectionParam;
+    
+    // use factory promise to get data
+    Issues.getIssues(issuesUrl).then(function(result) {
+      if (result.data.errors) {
+        //source error: do something
+      } else {
+        if (result.errors) {
+          //angular error: do something
+        } else {
+          // adding the issues to the scope
+          $scope.issues = result.data;
+          // adding special scopes for project keys, priorities, labels
+          // in case we want to make the filters <selects>
+          $scope.projects = _.uniq(_.pluck(result.data, 'projectKey'));
+          $scope.priorities = _.uniq(_.pluck(result.data, 'priority'));
+          var labelsPrep = _.compact(_.pluck(result.data, 'labels'));
+          // creating the label scope needs some massaging as they are an array
+          var labelsPrep2 = [];
+          $.each(labelsPrep, function() {
```

Comment:
If you are going to have labelsPrep and labelsPrep2, you may want to name the former labelsPrep1 or give the variable name more meaning.